### ISSUE TITLE
[Docs/Tooling] Default to os keyring backend

### ISSUE THREAD

**Issue by okdas** [pokt-network/poktroll#966]
## Objective

We've been using `test` as a keyring backend as this is the easiest backend for development purposes.
Now that we recommend users to utilize `poktrolld` binary in different ways, we should default to better security practices.

## Origin Document

<img width="1109" alt="image" src="https://github.com/user-attachments/assets/c59be85c-b91e-4cc9-b11a-d9a93d1f96fa">

## Goals

- Use `os` as the default keyring backend per [cosmos-sdk recommendations](https://docs.cosmos.network/main/user/run-node/keyring)
- Increase end-user security

## Deliverables

- [ ] Update CLI arguments to default to `os` keyring backend
- [ ] Update documentation/cheat-sheets where users interact with the `poktrolld` binary to remove the `--keyring-backend` argument (so it defaults to `os`)/

## Non-goals / Non-deliverables

- Do not remove `--keyring-backend=test` from the infrastructure. Infrastructure keyring considerations are different from user machines. Node operators should be familiar with their options for [keyring backends](https://docs.cosmos.network/main/user/run-node/keyring).

## General deliverables

- [ ] **Comments**: Add/update TODOs and comments alongside the source code so it is easier to follow.
- [ ] **Testing**: Add new tests (unit and/or E2E) to the test suite.
- [ ] **Makefile**: Add new targets to the Makefile to make the new functionality easier to use.
- [ ] **Documentation**: Update architectural or development READMEs; use [mermaid](https://mermaid-js.github.io/mermaid/) diagrams where appropriate.

---

**Creator**: @okdas
**Co-Owners**: TBD

**Comment by okdas** [comment]
@Olshansk this seems like a good first / community issue

**Comment by Olshansk** [comment]
@okdas Thanks for creating this. Let's keep it on the backlog and prioritize it when/if we get lots of people complaining about the CLI.

**Comment by okdas** [comment]
I believe the latest cosmos sdk allows us to change the `client.toml` defaults and template, so we can finally do this now.

**Comment by jorgecuesta** [comment]
@okdas is `os` a backend available on container environment? which is the most common practice to deploy products on the current days?

So far my last research when was dealing with keyring was that `os` is not supported on container environments.

**Comment by Olshansk** [comment]
_tl;dr No_

**os keyring needs a desktop credential manager (macOS Keychain, Windows CM, GNOME Keyring/KWallet via dbus).**

Those services don’t exist in normal Docker/Kubernetes images, so the backend just fails or hangs.

---

**What people actually do in containers**
- le backend → mount $HOME/.<app> from a Secret/PV and pass the key-password in an env var.
- test / memory → only for CI or localnet scripts (unencrypted).
- External signers → Tendermint KMS, cloud HSM/KMS, or a Ledger/YubiHSM sidecar if you want zero keys in the container.
- You can make os work by installing dbus + gnome-keyring and running a session bus inside the container, but that’s fragile and rarely worth the hassle.

---

**Bottom line**: use file in prod containers.

**Comment by jorgecuesta** [comment]
So, OS should not be the recommended right? also I do not see any of those
production ready keyring on base cosmos sdk, maybe this need research and
custom implementation of a well tested secure keyring?

**Comment by okdas** [comment]
@jorgecuesta I would recommend looking at how keys are handled by [cosmopilot](https://github.com/NibiruChain/cosmopilot) or [cosmos-operator](https://github.com/strangelove-ventures/cosmos-operator). These are good options, and even if not using kubernetes it might be a good idea to take patterns from these projects.

We haven't investigated these yet, but at some point I see us switching to a kubernetes operator in the future for internal infrastrtucture.

**Comment by jorgecuesta** [comment]
Thanks but we need to think tbat they may only work for node key (1) but
relay miner need to work with hundred of them.
